src/esthri/src/blocking.rs

```diff
@@ -19,69 +20,91 @@ use super::Result;
 use super::SyncParam;
 
 #[tokio::main]
-pub async fn head_object<T>(s3: &T, bucket: &str, key: &str) -> Result<Option<ObjectInfo>>
+pub async fn head_object<T, SR0, SR1>(s3: &T, bucket: SR0, key: SR1) -> Result<Option<ObjectInfo>>
 where
     T: S3 + Send,
+    SR0: AsRef<str>,
```

Comment:
Is this the best way to do this?  Would like to be able to pass a &str and String for either or both of the bucket/key param.... using one type parameter seems to require that they both be the same concrete type.

Reply:
Yup I believe that's correct